app/models/user.rb

```diff
@@ -10,6 +10,8 @@ class User < ActiveRecord::Base
 
   has_many :assignments, foreign_key: :submitter_id
 
+  has_many :comments, through: :assignments
```

Comment:
Since comments will always be sorted by date, you could add a `-> {  order(...) }` block here instead of adding sorts manually like in `assignment_for(user).try(&:comments).try(:order, 'date') || []`

That way you will never forgot to add the sort message